src/main/java/com/faforever/api/clan/ClanService.java

```diff
@@ -120,11 +128,12 @@ void acceptPlayerInvitationToken(String stringToken, Authentication authenticati
     Clan clan = clanRepository.findById(clanId)
       .orElseThrow(() -> new ApiException(new Error(ErrorCode.CLAN_NOT_EXISTS, clanId)));
 
-    Player newMember = playerRepository.findById(invitation.getNewMember().getId())
-      .orElseThrow(() -> new ProgrammingError("ClanMember does not exist: " + invitation.getNewMember().getId()));
+    Player newMember = clan.getIsOpen() ? null :
+      playerRepository.findById(invitation.getNewMember().getId())
+        .orElseThrow(() -> new ProgrammingError("ClanMember does not exist: " + invitation.getNewMember().getId()));
 
 
-    if (player.getId() != newMember.getId()) {
+    if (!invitation.getClan().getIsOpen() && player.getId() != newMember.getId()) {
```

Comment:
Throw a different custom ApiException with your own error here. Otherwise test will be missleading